setup.md

```diff
@@ -4,166 +4,241 @@ This document will guide you through setting up Fennel on your
-Fennel scripts to automate your application. However, if you are
-working with an application that has more restrictions, it may be
-simpler to compile your Fennel code to Lua during the build process
-and only include the Lua output in the application.
+This section consists of the following subsections:
 
-## Embedding the compiler
+* [Embedding the Fennel compiler in a Lua application](#embedding-the-fennel-compiler-in-a-lua-application)
+* [Performing ahead-of-time compilation](#performing-ahead-of-time-compilation)
 
-Add `fennel.lua` to your repository, then you can load it from Lua like so:
+**Note**: Embedding the Fennel compiler in an application is the more
+flexible option, and is recommended. By embedding the Fennel compiler
+in an application, users can write their own extension scripts in
+Fennel to interact with the application. If the application is more
+restricted, then compiling Fennel code to Lua during the build
+process, and including the Lua output in the application may be
+easier.
+
+## Embedding the Fennel compiler in a Lua application
+
+The Fennel compiler can be added to your code repository, and then
+loaded from Lua. This allows you to call functions defined in Fennel
+code from Lua, and functions defined in Lua from Fennel without
+slowdown.
+
+### To embed the Fennel compiler in a Lua application
+
+1. Add `fennel.lua` to your code repository
+2. Add the following lines to your Lua code:
 
 ```lua
 local fennel = require("fennel")
 table.insert(package.loaders or package.searchers, fennel.searcher)
 local mylib = require("mylib") -- will compile and load code in mylib.fnl
 ```
 
-You can call any function defined in Fennel code from Lua with zero
-overhead, and vice versa.
+**Note**: Optionally, if you want the Fennel REPL to print tables and
+other data in a more readable format, you can add `fennelview.fnl` to
+your code repository. For more helpful compiler errors, you can add
+`fennelfriend.fnl` to your code repository.
+
+## Performing ahead-of-time compilation
 
-In order to get the repl to be able to print tables correctly, you
-probably also want to add `fennelview.fnl`. In order to improve the
-compiler errors, you can add `fennelfriend.fnl`. However, both these
-modules are optional.
+If the target system of your application does not have Fennel
+installed, but has Lua installed, Fennel offers ahead-of-time
+compilation. This allows you to compile `.fnl` files to `.lua` files
+before shipping an application.
 
-## Ahead of time compilation
+This section will guide you through creating a `Makefile` to perform
+this compilation for you.
 
-If you need to ship `.lua` files in your program, you can use `make`
-to perform the compilation. Add this to your `Makefile`:
+### To perform ahead-of-time compilation
+
+1. Run `touch Makefile`
+2. Add the following lines to the `Makefile`:
 
 ```
 %.lua: %.fnl fennel
-	./fennel --compile $< > $@
+    ./fennel --compile $< > $@
```

Comment:
Most of the time it's OK to replace tabs with spaces; however in this context (inside a Makefile) it will actually cause a syntax error. =( This is one of those things that's just really stupid about `make`, which is overall pretty good.

Reply:
Damnit, I'm not used to having tabs in my Markdown files. I totally didn't realize my editor did this. Thanks for catching this! haha, I will fix this!